home/import_content_pages.py

```diff
@@ -57,11 +47,25 @@ def __init__(
         self.progress_queue = progress_queue
         self.purge = purge in ["True", "yes", True]
         self.locale = locale
-        self.shadow_pages: dict[str, ShadowContentPage] = {}
+        self.locale_map: dict[str, Locale] = {}
+        self.shadow_pages: dict[PageId, ShadowContentPage] = {}
         self.go_to_page_buttons: dict[
-            str, dict[int, list[dict[str, Any]]]
+            PageId, dict[int, list[dict[str, Any]]]
         ] = defaultdict(lambda: defaultdict(list))
 
+    def locale_from_display_name(self, langname: str) -> str:
```

Comment:
Isn't this returning a `Locale`? And is there a reason for managing our own map vs using the `cache` decorator?